skopt/dummy_opt.py

```diff
@@ -52,14 +71,45 @@ def dummy_minimize(func, dimensions, n_calls=100, random_state=None):
     """
     rng = check_random_state(random_state)
     space = Space(dimensions)
-    X = space.rvs(n_samples=n_calls, random_state=rng)
+    if x0 is None:
+        x0 = []
+    elif not isinstance(x0[0], list):
+        x0 = [x0]
+
+    if not isinstance(x0, list):
+        raise ValueError("Expected x0 to be a list, but got %s" % type(x0))
 
-    init_y = func(X[0])
+    if y0 is None and x0:
+        y0 = [func(x) for x in x0]
+        n_func_calls = len(y0)
+    elif x0:
+        if isinstance(y0, Iterable):
+            y0 = list(y0)
+        elif isinstance(y0, numbers.Number):
+            y0 = [y0]
+        else:
+            raise ValueError(
+                "Expected y0 to be an iterable or a scalar, got %s" % type(y0))
+        if len(x0) != len(y0):
+            raise ValueError("x0 and y0 should have the same length")
+        if not all(map(np.isscalar, y0)):
+            raise ValueError(
+                "y0 elements should be scalars")
+        n_func_calls = 0
+    else:
+        y0 = []
+        n_func_calls = 0
+
+    X_left = space.rvs(n_samples=n_calls - n_func_calls, random_state=rng)
+    init_y = func(X_left[0])
```

Comment:
That's a border case, but if `len(x0) == n_calls` and `y0 is None`, then you evaluate `func` once more than allowed.